File: src/main/java/com/dozycoffee/wms/warehouse/adapter/in/web/response/WorkAreaResponse.java
package com.dozycoffee.wms.warehouse.adapter.in.web.response;

import com.dozycoffee.wms.warehouse.application.port.in.result.WorkAreaResult;
import com.dozycoffee.wms.warehouse.domain.enumeration.AreaCode;
import com.dozycoffee.wms.warehouse.domain.enumeration.AvailabilityStatus;

public record WorkAreaResponse(
        Long workAreaId,
        Long warehouseId,
        AreaCode areaCode,
        String areaName,
        int maxCapacity,
        int usedCapacity,
        AvailabilityStatus workAreaStatus
) {

    public static WorkAreaResponse from(WorkAreaResult result) {
        return new WorkAreaResponse(
                result.workAreaId(),
                result.warehouseId(),
                result.areaCode(),
                result.areaName(),
                result.maxCapacity(),
                result.usedCapacity(),
                result.workAreaStatus()
        );
    }
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/adapter/in/web/response/WorkAreaResponse.kt
package com.dozycoffee.wms.warehouse.adapter.`in`.web.response

import com.dozycoffee.wms.warehouse.application.port.`in`.result.WorkAreaResult
import com.dozycoffee.wms.warehouse.domain.enumeration.AreaCode
import com.dozycoffee.wms.warehouse.domain.enumeration.AvailabilityStatus

data class WorkAreaResponse(
    val workAreaId: Long,
    val warehouseId: Long,
    val areaCode: AreaCode,
    val areaName: String,
    val maxCapacity: Int,
    val usedCapacity: Int,
    val workAreaStatus: AvailabilityStatus
) {
    companion object {
        fun from(result: WorkAreaResult): WorkAreaResponse {
            return WorkAreaResponse(
                result.workAreaId,
                result.warehouseId,
                result.areaCode,
                result.areaName,
                result.maxCapacity,
                result.usedCapacity,
                result.workAreaStatus
            )
        }
    }
}
